viewing_party/party.py

```diff
@@ -1,23 +1,157 @@
+
+    return friends_watched_list
+
+def get_unique_watched(user_data):
+    """ Return a list of movies only the user has watched. """
+    user_watched = user_movies_watched(user_data)
+    friends_watched = friends_movies_watched(user_data)
+    user_unique_watched = []
+
+    for movie in user_watched:
+        if not movie in friends_watched:
+            user_unique_watched.append(movie)
+
+    return user_unique_watched
+
+def get_friends_unique_watched(user_data):
+    """ Return a list of movies only the user's friends have watched. """
+    user_watched = user_movies_watched(user_data)
+    friends_watched = friends_movies_watched(user_data)
+    friends_unique_watched = []
+
+    for movie in friends_watched:
+        if not movie in user_watched and not movie in friends_unique_watched:
+            friends_unique_watched.append(movie)
+
+    return friends_unique_watched
+
+# def friends_movies(user_data):
+#     friends_movie_data = []
+#     for movies_dict in user_data["friends"]:
+#         for movies in movies_dict["watched"]: 
+#             friends_movie_data.append(movies) 
+#     return (friends_movie_data)
+
+# def get_unique_watched(user_data):
+#     user_unique_movies =[]
+#     for movie in user_data["watched"]:
+#         if movie not in friends_movies(user_data):
+#             user_unique_movies.append(movie)
+#     return(user_unique_movies)
+
+# def get_friends_unique_watched(user_data):
+#     friends_unique_movies = []
+#     filter_out_dupes = []
+#     for movie in friends_movies(user_data):
+#         if movie not in user_data["watched"]:
+#             friends_unique_movies.append(movie)
+
+#     for movie in friends_unique_movies:
+#         if movie not in filter_out_dupes:
+#             filter_out_dupes.append(movie)
+#     return (filter_out_dupes)
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    """ Return a list of recommended movies. """
+    recommended_movies = []
+    for movie in get_friends_unique_watched(user_data):
+        if movie["host"] in user_data["subscriptions"]:
+            recommended_movies.append(movie)
+    return recommended_movies
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    rec_based_on_genre = []
+    most_freq_genre = get_most_watched_genre(user_data)
+    for movie in get_friends_unique_watched(user_data):
+        if movie["genre"] == most_freq_genre:
+            rec_based_on_genre.append(movie)
+    return rec_based_on_genre
+
+def get_rec_from_favorites(user_data):
+    rec_based_on_favorites = []
+    for movie in get_unique_watched(user_data):
+        if movie in user_data["favorites"]:
+            rec_based_on_favorites.append(movie)
+        else:
+            rec_based_on_favorites = []

```

Comment:
No need to use this `else` here since you probably wouldn't want to reassign the `rec_based_on_favorites` variable to an empty list. We aren't testing a case that would show this bug, but it would be one like:
```python
    test_data = {
        "favorites": [INTRIGUE_1, FANTASY_3b],
        "watched": [
            FANTASY_1b,
            ],
        "friends":  [
            {
                "watched": [
                    FANTASY_3b,
                ]
            }
        ]
    }
```
I think in this case, `INTRIGUE_1` should be returned as a recommendation, correct? This function would return an empty list in this case. Could be something worth investigating!